src/internal/operations/restore.go

```diff
@@ -129,13 +129,13 @@ func (op *RestoreOperation) Run(ctx context.Context) (restoreDetails *details.De
 	// -----
 
 	ctx, end := diagnostics.Span(ctx, "operations:restore:run")
-	defer func() {
-		end()
-	}()
+	defer end()
 
 	ctx, flushMetrics := events.NewMetrics(ctx, logger.Writer{Ctx: ctx})
 	defer flushMetrics()
 
+	ctx = clues.AddTrace(ctx)
```

Comment:
These changes look unrelated to the PR. Could you please add them as tech debt items for future PRs? :) Makes it a little harder to review.

Reply:
Leaving these in for now.  I'll try to be more mindful in the future.